src/IIIFPresentation/API/Features/Storage/Requests/PostHierarchicalCollection.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Data;
+using API.Auth;
+using API.Converters;
+using API.Features.Storage.Helpers;
+using API.Helpers;
+using API.Infrastructure.Requests;
+using API.Settings;
+using AWS.S3;
+using AWS.S3.Models;
+using Core;
+using IIIF.Presentation.V3;
+using IIIF.Presentation.V3.Content;
+using IIIF.Serialisation;
+using MediatR;
+using Microsoft.Extensions.Options;
+using Models.API.General;
+using Repository;
+using Repository.Helpers;
+using DatabaseCollection = Models.Database.Collections;
+using IIdGenerator = API.Infrastructure.IdGenerator.IIdGenerator;
+
+namespace API.Features.Storage.Requests;
+
+public class PostHierarchicalCollection(
+    int customerId,
+    string slug, 
+    UrlRoots urlRoots,
+    string rawRequestBody) : IRequest<ModifyEntityResult<Collection, ModifyCollectionType>>
+{
+    public int CustomerId { get; } = customerId;
+
+    public string Slug { get; } = slug;
+    
+    public UrlRoots UrlRoots { get; } = urlRoots;
+    
+    public string RawRequestBody { get; } = rawRequestBody;
+}
+
+public class PostHierarchicalCollectionHandler(
```

Comment:
General comment/personal opinion: this method is doing a lot and there's a lot of vars that are used for single purposes. Could it be split up to make it easier to read at a glance?

```cs
var parentCollection = GetParentCollection(request);
if (parentCollection == null) return ErrorHelper.NullParentResponse<Collection>();

var id = GetUniqueId();
if (id == null) return ErrorHelper.CannotGenerateUniqueId<Collection>();

var collection = BuildCollection(request, id, parentCollection);
if (collection == null) return ErrorHelper.CannotValidateIIIF<Collection>();

await SaveStuff()...
```